### PR TITLE
rake: Add administrative tasks

### PR DESCRIPTION
**Pull request by ethan-readyset**

### CODE REVIEW

File: lib/tasks/readyset.rake [hunk 1]
```diff
@@ -1,13 +1,87 @@
...
+
+  namespace :proxied_queries do
+    desc 'Creates caches for all of the supported queries on ReadySet'
+    task cache_all_supported: :environment do
+      Readyset::Query::ProxiedQuery.cache_all_supported!
+    end
+
+    desc 'Clears the list of proxied queries on ReadySet'
+    task drop_all: :environment do
+      Readyset.raw_query('DROP ALL PROXIED QUERIES')
```
Comment (helpotters) [inline lib/tasks/readyset.rake:44]: **_thought:_** We should look into [frozen string literals](https://www.mintbit.com/blog/exploring-frozen-string-literal-a-ruby-magic-comment). :snowflake:

File: lib/tasks/readyset.rake [hunk 2]
```diff
@@ -1,13 +1,87 @@
...
+        Readyset.drop_cache!(args[:name])
+      else
+        puts 'A cache name must be passed to this task'
+      end
+    end
+
+    desc 'Drops all the caches on ReadySet'
+    task drop_all: :environment do
+      Readyset::Query::CachedQuery.drop_all!
+    end
```
Comment (helpotters) [inline lib/tasks/readyset.rake:81]: **_suggestion:_** Do you think these tasks would be considered 'dangerous' despite being copies of data?

Potential safety measure: adding a confirmation prompt and comparing it against a '`confirm_token`'. [Reference article with some neat ideas.](https://www.lynnbright.com/6-tips-to-ensure-your-rake-task-runs-smoothly/)

File: lib/tasks/readyset.rake [hunk 3]
```diff
@@ -1,13 +1,87 @@
...
 require 'progressbar'
+require 'terminal-table'
 
 namespace :readyset do
+  desc 'Creates a cache from the given query ID'
+  task :create_cache, [:id] => :environment do |_, args|
+    if args.key?(:id)
+      Readyset.create_cache!(id: args[:id])
+    else
+      puts 'A query ID must be passed to this task'
```
Comment (helpotters) [inline lib/tasks/readyset.rake:12]: **question:_** Should we use `Rails.logger` in addition to the terminal messages?

File: lib/tasks/readyset.rake [hunk 4]
```diff
@@ -69,4 +141,22 @@ namespace :readyset do
...
+
+  desc 'Prints information about the tables known to ReadySet'
+  task tables: :environment do
+    rows = Readyset.raw_query('SHOW READYSET TABLES').
+      map { |result| [result['table'], result['status'], result['description']] }
+    table = Terminal::Table.new(headings: [:table, :status, :description], rows: rows)
+
+    puts table
+  end
 end
```
Comment (helpotters) [inline lib/tasks/readyset.rake]: :sunglasses: : I can definitely see these rake tasks getting a lot of use. Nice, nice, nice.

File: lib/tasks/readyset.rake (hunk 1 above)
Comment (ethan-readyset) [reply lib/tasks/readyset.rake:44]: Agreed! I'll update the literals in this file

**Comment by ethan-readyset** [comment]
@helpotters Ready for another pass

File: spec/rake_spec.rb [hunk 5]
```diff
@@ -10,17 +10,108 @@
...
+          | id                 | name               | text                            | always | count |
+          +--------------------+--------------------+---------------------------------+--------+-------+
+          | q_4f3fb9ad8f73bc0c | q_4f3fb9ad8f73bc0c | SELECT                          | false  | 0     |
+          |                    |                    |   "public"."cats"."breed"       |        |       |
+          |                    |                    | FROM                            |        |       |
+          |                    |                    |   "public"."cats"               |        |       |
+          |                    |                    | WHERE                           |        |       |
+          |                    |                    |   ("public"."cats"."name" = $1) |        |       |
+          +--------------------+--------------------+---------------------------------+--------+-------+
+        TABLE
```
Comment (helpotters) [inline spec/rake_spec.rb:71]: **_nitpick (100% skippable):_** This kind of reminds me of a feature spec (interactive, user-facing expectation). Non-specific, bare-minimum matchers could help with spec brittleness. Such as: "I expect these values to be in there somewhere" and "I expect these columns to be there somewhere". 


```
  scenario 'send shipment' do
    # note: `have_content` means *anywhere* on the webpage
    expect(page).to have_content('Ready to ship')

    within '.qa-passed' do
      click_button 'Add to shipment'
    end
    
    within '.factory-second' do
      click_button 'Add to shipment'
    end
    
    within '.shipping-list' do
      click_button 'Send shipment'
    end

    expect(page).to have_content("Robots #{robot1.id}, #{robot2.id} is shipped successfully.")
  end
  ```

File: spec/rake_spec.rb [hunk 6]
```diff
@@ -10,17 +10,108 @@
...
+              to output("A cache name must be passed to this task\n").to_stdout
+          end
+        end
+      end
+
+      describe 'drop_all' do
+        it 'removes all the caches on ReadySet' do
+          build_and_create_cache(:cached_query)
+          build_and_create_cache(:cached_query_2)
+
```
Comment (helpotters) [inline spec/rake_spec.rb:100]: :sunglasses: Your helper methods have gotten really good.

**Review by helpotters — APPROVED** [review]
Non-blocking questions and comments. Merge at your discretion!

File: lib/tasks/readyset.rake (hunk 2 above)
Comment (ethan-readyset) [reply lib/tasks/readyset.rake:81]: I think that's a great idea! But I will probably leave it as-is for now in the interest of time

File: lib/tasks/readyset.rake (hunk 3 above)
Comment (ethan-readyset) [reply lib/tasks/readyset.rake:12]: Yup, I think you're right! I'll update that

File: spec/rake_spec.rb (hunk 5 above)
Comment (ethan-readyset) [reply spec/rake_spec.rb:71]: I like that idea! I think for now, I'll keep things as-is in the interest of time, but this is a prime candidate for refactoring in the future!

**Pull request merged**